test/optimize/test_optimize.py

```diff
@@ -199,3 +200,14 @@ def check_minimize(objective, library, solver, allow_failed_starts=False):
     ]:
         assert np.isfinite(result.optimize_result.list[0]['fval'])
         assert result.optimize_result.list[0]['x'] is not None
+
+
+def test_MPIPool_Engine():
+    """
+    Test the MPIPoolEngine by calling an example script with mpiexec.
+    """
+    # get the path to this file:
+    path = os.path.dirname(__file__)
+    # run the example file.
+    os.system(f"mpiexec -np 2 python -m mpi4py.futures"
+              f" {path}../../doc/example/example_MPIPool.py")
```

Comment:
Can this test ever fail? Please check the return code. Would suggest to use `subprocess`.

Reply:
Tried it with subprocess (smhw did not work). Now os.sytems() return is saved and an error is raised if it is not a 0.

Reply:
Must work with subprocess. If the return code is checked - ok. Not sure if output is shown if launched with subprocess.

Reply:
So on my computer i can call the example_MPIPool.py with the os and it actually returns 0 if it runs without problems and 256 otherwise. But in the test on github it returns 256 with 
```
Primary job  terminated normally, but 1 process returned
a non-zero exit code.. Per user-direction, the job has been aborted.
-------------------------------------------------------
--------------------------------------------------------------------------
mpiexec detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:
```
but as far as i could read the implementation of `mpi4py.futures.MPIPoolExecutor`, it automatically performs a shutdown if you use with a `with` statement. any Idea what the problem could be?

Reply:
> . But in the test on github it returns 256 with [...]

From what I can see, `mpi4py` is simply not installed

```
----------------------------- Captured stderr call -----------------------------
/opt/hostedtoolcache/Python/3.8.7/x64/bin/python: Error while finding module specification for 'mpi4py.futures' (ModuleNotFoundError: No module named 'mpi4py')
/opt/hostedtoolcache/Python/3.8.7/x64/bin/python: Error while finding module specification for 'mpi4py.futures' (ModuleNotFoundError: No module named 'mpi4py')
```